client/src/pages/login.js

```diff
@@ -48,8 +48,16 @@ export class Login {
       await this.auth.login(this.username, this.password);
       this.router.navigateToRoute("index");
     } catch (e) {
+      let loginErrorMessage = "";
+      if (e.status === 401) {
+        loginErrorMessage = this.i18n.tr("general.loginFailed");
```

Comment:
Can you assign to `this.loginError` directly in this `if/else` construct? There is no real need for another variable `loginErrorMessage ` here imo.